### DISCUSSION TITLE
Multifuel reaction/Ramp activation time

### DISCUSSION THREAD

**Discussion by MartonTimea** [firemodels/fds#12907] [Q&A] [answered]
[hrr_multifuel_reaction_test.txt](https://github.com/firemodels/fds/files/15261638/hrr_multifuel_reaction_test.txt)
Hi there,

I have successfully implemented a multifuel reaction in my model, using two different fire ramps, based on when the sprinkler is estimated to activate. (estimation made by hand calculation).

Now, i am looking into doing the same, but using the HRR Freeze option, based on LINK TEMPERATURE activation time of the sprinklers in the model. I know how to implement the freeze function for a single fire ramp, but I am wondering if there is anyway, to have it implemented for a multifuel reaction. I am not sure how to define the second ramp ('FIRE RAMP2'). Is there anyway to have the 2nd ramp start when the first ramp freezes?

I have attached the fds code for my test case, in case it is needed.  Thank you beforehand.

&RAMP ID='FIRE_RAMP1'    T=  0.0     F=  0       DEVC_ID=    'Freeze Time'/  
&RAMP ID='FIRE_RAMP1'    T=  73.0    F=  0.03    /   250 kW
&RAMP ID='FIRE_RAMP1'    T=  154.0   F=  0.11    /   1115    kW
&RAMP ID='FIRE_RAMP1'    T=  208.0   F=  0.20    /   2033    kW
&RAMP ID='FIRE_RAMP1'    T=  275.0   F=  0.36    /   3554    kW
&RAMP ID='FIRE_RAMP1'    T=  329.0   F=  0.51    /   5087    kW
&RAMP ID='FIRE_RAMP1'    T=  380.0   F=  0.68    /   6787    kW
&RAMP ID='FIRE_RAMP1'    T=  429.0   F=  0.86    /   8650    kW
&RAMP ID='FIRE_RAMP1'    T=  461.3   F=  1.00    /   10000   kW
&RAMP ID='FIRE_RAMP1'    T=  900.0   F=  1.00    /   10000   kW
                            
&RAMP ID='FIRE_RAMP2'    T=  0       F=  0       /   
&RAMP ID='FIRE_RAMP2'    T=  x       F=  0       /   
&RAMP ID='FIRE_RAMP2'    T=  x+1     F=  ?       /

**Comment by drjfloyd** [comment]
Do you have a single burning object where the second fuel starts with sprinkler operation or multiple burning objects some with the first fuel and some with the second fuel?

**Reply by MartonTimea** [reply]
Single burner object. But if it is easier to implement with two burning objects, that could work too.

**Reply by drjfloyd** [reply, marked as answer]
Two objects will be the easiest. All you would need to do is create the second object when sprinklers activate or have the second object present and create a burner ontop of it when sprinklers activate.  Then the second fire ramp will start once the object/vent is created.

You can do it with one object but it will take a little more effort. Instead of using HRRPUA for the SURF you can use MASS_FLUX,SPEC_ID, and RAMP_MF.  You will need to convert your HRRPUA curves to mass flux curves by dividing by each fuel's heat of combustion (which you can get from the .out file of the case you ran before). Fuel 1 you use the same DEVC_ID for the RAMP.  For fuel 2 what you want is the T value to start counting up from zero when the sprinkler operates. You can do that by having two TIME DEVC. One that freezes (i.e. your current Freeze Time), and one that does not.  Then use a subtract CTRL to subtract one from the other. Until the device freezes both will have the same time and the subtraction will result in 0. Once the freeze happens, the non-frozen time will keep increasing and not the subtraction will  give time since sprinkler operation. This is what you need for T for the second ramp and can assign that ramp the CTRL_ID of the subtraction.

**Reply by MartonTimea** [reply]
Thank you very much. This is clear enough. Much appreciated.

**Reply by MartonTimea** [reply]
If I use two objects tough, I don't see how I can assign the HRR to RAMP2, based on the HRR reached at RAMP1, when the sprinklers activated. Is there a way to do that? Basically, I want the HRR to freeze at sprinkler activation, but to have a different soot yield.

**Reply by drjfloyd** [reply]
I don't know what exactly you are trying to do. Please give a detailed description of how you want the two ramps to behave with respect to sprinkler operation including fuels, yields, and the process you wish to apply to define the HRRPUA.

**Comment by LordMicidial** [comment]
I recall this thread post a similar question that hopefully could help MartonTimea and let me understand if my reasoning are adequate:

&PROP ID='Generic Commercial Link', QUANTITY='SPRINKLER LINK TEMPERATURE', ACTIVATION_TEMPERATURE=78.0/
&CTRL ID='ShutDown', FUNCTION_TYPE='TIME_DELAY', DELAY=10.0, LATCH=.FALSE., INPUT_ID='SLINK'/
&CTRL ID='SIMCTRL', LATCH=.FALSE., FUNCTION_TYPE='KILL', INPUT_ID='ShutDown'/
&DEVC ID='SLINK', PROP_ID='Generic Commercial Link', XYZ=117.5,71.13,6.2/
&DEVC ID='HRRTime', QUANTITY='TIME', XYZ=0.0,0.0,0.0, NO_UPDATE_DEVC_ID='SLINK'/

This above is the code I use to introduce a commercial sprinkler link at a selected location and a Time device "HRRTime" which stop counting when the Slink reach 78°.

When running, the code related to the RAMP_Qs might be associated to a Time device as the "HRRTime" I've designed as per input below.

&RAMP ID='BURNER_RAMP_Q', T=0.0, F=0.0, DEVC_ID='HRRTime'/
&RAMP ID='BURNER_RAMP_Q', T=2.0, F=1.5008E+4/
&RAMP ID='BURNER_RAMP_Q', T=4.0, F=6.0032E+4/

My idea to answer MartonTimea is to proposing this solution setting "DEVC_ID='HRRTime'/" in each of the RAMP_Q associated to the other objects that will take fire.

Actually, I'm here to seek confirmation of what stated below:

The second SURF (mob) is located close to the burner (Innesco). Given the fact that at some time the HRR curve will be frozen by the HRRtime device and the "Mob" surface will start to ignite at 275°, using the same HRRtime device for two SURF (and HRR curves) is appropriate? Shall I undertake another reasoning?


&SURF ID='Innesco',
      RGB=255,102,0,
      HRRPUA=1000.0,
      RAMP_Q='Innesco_RAMP_Q',
 THICKNESS(1)=0.2,
      MATL_ID(1,1)='Wood'/
&RAMP ID='Innesco_RAMP_Q', T=0.0, F=0.0, DEVC_ID='HRRTime'/
&RAMP ID='Innesco_RAMP_Q', T=50.0, F=0.01/
&RAMP ID='Innesco_RAMP_Q', T=100.0, F=0.04/
&RAMP ID='Innesco_RAMP_Q', T=150.0, F=0.1/
&RAMP ID='Innesco_RAMP_Q', T=200.0, F=0.17/
&RAMP ID='Innesco_RAMP_Q', T=250.0, F=0.27/
&RAMP ID='Innesco_RAMP_Q', T=300.0, F=0.38/
&RAMP ID='Innesco_RAMP_Q', T=350.0, F=0.52/
&RAMP ID='Innesco_RAMP_Q', T=400.0, F=0.68/
&RAMP ID='Innesco_RAMP_Q', T=450.0, F=0.86/
&RAMP ID='Innesco_RAMP_Q', T=4844.0, F=1.0/
&RAMP ID='Innesco_RAMP_Q', T=1.1921E+4, F=1.0/
&RAMP ID='Innesco_RAMP_Q', T=1.9371E+4, F=0.0/
&SURF ID='Mob',
      RGB=255,255,16,
      HRRPUA=7500.0,
      RAMP_Q='Mob_RAMP_Q',
      IGNITION_TEMPERATURE=275.0,
 THICKNESS(1)=0.2,      
MATL_ID(1,1)='Wood',
      MATL_MASS_FRACTION(1,1)=1,0,
&RAMP ID='Mob_RAMP_Q', T=0.0, F=0.0,DEVC_ID='HRRTime'/
&RAMP ID='Mob_RAMP_Q', T=50.0, F=0.02/
&RAMP ID='Mob_RAMP_Q', T=100.0, F=0.06/
&RAMP ID='Mob_RAMP_Q', T=150.0, F=0.14/
&RAMP ID='Mob_RAMP_Q', T=200.0, F=0.25/
&RAMP ID='Mob_RAMP_Q', T=250.0, F=0.38/
&RAMP ID='Mob_RAMP_Q', T=300.0, F=0.55/
&RAMP ID='Mob_RAMP_Q', T=350.0, F=0.75/
&RAMP ID='Mob_RAMP_Q', T=400.0, F=0.98/
&RAMP ID='Mob_RAMP_Q', T=4033.0, F=1.0/
&RAMP ID='Mob_RAMP_Q', T=1.1161E+4, F=1.0/
&RAMP ID='Mob_RAMP_Q', T=1.8422E+4, F=0.0/

**Comment by drjfloyd** [comment]
This line will stop FDS 10 s after the sprinkler operates:
&CTRL ID='SIMCTRL', LATCH=.FALSE., FUNCTION_TYPE='KILL', INPUT_ID='ShutDown'/

I don't think that is what @MartonTimea is looking to do.  It is not clear to me exactly what he is looking to do, hence my last post which has not yet been responded to. Without being confident that I understand what the desired behavior is, I cannot comment on whether or not your approach is a potential solution.
